### PR TITLE
add unit tests for the enumerables

### PR DESCRIPTION
**Pull request by Bluette1**
## In this project we: 
- added specs for all public methods previously created.
- Initialized RSpec.
- Each method was tested with the describe block.
- Used one "it" block for each expected behavior.
- Used let.
- Tested positive and negative scenarios

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,244 @@
+require_relative '../enumerable'
+
+describe 'enumerable' do
+  describe '#my_each' do
+    let(:item) { [1, 2, 3, 4] }
```
Comment (Forison) [inline spec/enumerable_spec.rb]: - Kindly move this ```let(:item) { [1, 2, 3, 4] }``` line and place it between lines 3 and 4 thus making it a global variable. Then, remove it from all other places they currently exist.

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      expect(item.my_select.to_a).to match_array(item.select.to_a)
+    end
+
+    it 'returns the expected result when passed a block' do
+      expect(item.select(&:even?)).to match_array(item.my_select(&:even?))
+    end
+  end
+
+  describe '#my_all' do
+    let(:item) { %w[ant bear cat] }
```
Comment (Forison) [inline spec/enumerable_spec.rb]: - Kindly move this ```let(:item) { %w[ant bear cat] }``` line and place it between lines 3 and 4 thus making it a global variable. Then, remove it from all other places they currently exist.

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    end
+
+    let(:item) { %w[ant beat cat] }
+    let(:contexts) { [/t/, '*', 'beat', 'BOOM/'] }
+    it 'returns the expected result for a regex test' do
+      contexts.each do |context|
+        expect(item.my_any?(context)).to eq(item.any?(context))
+      end
+    end
+    let(:contexts) { [[[1, 2i, 3.14], %w[ant bear cat], [nil, true, 99]], [Numeric, String, Integer]] }
```
Comment (Forison) [inline spec/enumerable_spec.rb]: Do same for ```let(:contexts) { [[[1, 2i, 3.14], %w[ant bear cat], [nil, true, 99]], [Numeric, String, Integer]] } ```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes required❌
# 🙋‍♂️ Team!!!
#### You have done a great on this one, I am impressed at the level of your test, however there some behaviors which are inconsistent with the original Ruby method.
#### my_each test
* [ ] This test is failing ```expect(item.my_each{|x| x}).to eql(item.each{|x| x})``` check the expectation in the image below
EX1
you could verify it from the behavior of the method
```p [1, 2, 3, 4].each{|x| x}``` should return same results as  ```p [1, 2, 3, 4].my_each{|x| x}```

#### my_each_with_index test
* [ ] This test is failing ```expect(item.my_each_with_index{|x| x}).to eql(item.each{|x| x})``` check the expectation in the image below
ex2
you could verify it from the behavior of the method
```p [1, 2, 3, 4].each_with_index{|x| x}``` should return same results as 
```p [1, 2, 3, 4].my_each_with_index{|x| x}```
* [ ]  I will point out a few things below👇 that you should do to refactor your tests to save space as well as observing the DRY rule of Ruby🤗.
* [ ] Add a "Getting started" to the readme, in that section describe how to set up the project on a local machine. This should come to complement the part when you indicated  run```rspec```.

**[OPTIONAL]**
Add a description to the repository, it makes your repository more professional. You will also be asked to do this when you are about to graduate.

## That is it, kindly make the change and submit a re-review🤠 .
#### Happy coding🤖.
Addo Forison

**Comment by Bluette1** [comment]
@Forison Thank you very much for your feedback!

**Review by rOluochKe — APPROVED** [review]
Hello Team!
### You have done great and for also making changes from the previous review :clap: The test passes and the project has been approved. Congratulations :boom: :100:
### You can merge and submit a project completion form from your dashboard. Good luck with the next project.
- Happy coding!

**Comment by Bluette1** [comment]
@rOluochKe Thank you very much  for your review and feedback!

**Pull request merged**